### PR TITLE
NCSDK-22152: Add manifest representation

### PR DESCRIPTION
**Pull request by tomchy**
This PR is a part of a bigger effort of allowing to process dependency manifests.

### CODE REVIEW

File: include/suit_types.h [hunk 1]
```diff
@@ -15,11 +15,14 @@
...
 #define SUIT_MAX_NUM_COMPONENT_ID_PARTS 5  ///! The maximum number of bytestrings in a component ID.
-#define SUIT_MAX_NUM_COMPONENTS 4  ///! The maximum number of components referenced in the manifest.
+#define SUIT_MAX_NUM_COMPONENTS 6  ///! The maximum number of components referenced in the manifest.
+#define SUIT_MAX_NUM_COMPONENT_PARAMS (SUIT_MAX_NUM_COMPONENTS * SUIT_MAX_MANIFEST_DEPTH) ///! The maximum number of active components during processing dependency manifests.
+#define SUIT_MAX_NUM_INTEGRATED_PAYLOADS 5  ///! The maximum number of integrated payloads in a single manifest.
 #define SUIT_MAX_COMMAND_ARGS 3  ///! The maximum number of arguments consumed by a single command.
 #define SUIT_SUIT_SIG_STRUCTURE1_MAX_LENGTH 55  ///! The maximum length of the Sig_structure1 structure. Current value allows to store only 256-bit long digests.
 #define SUIT_MAX_SEQ_DEPTH 5  ///! The maximum number of command sequences that may be encapsulated.
 #define SUIT_SEQ_EXEC_DEFAULT_STATE 0  ///! The default value of the cmd_exec_state.
+#define SUIT_MAX_MANIFEST_DEPTH 3 ///! The maximum nesting level of hierarhical manifests.
```
Comment (robertstypa) [inline include/suit_types.h:25]: hierarhical -> hierarchical

**Review by robertstypa — APPROVED** [review]

**Review by adsz-nordic — COMMENTED** [review]
Please check my findings.

File: include/suit_processor.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	struct zcbor_string payload;
+};
+
+struct suit_manifest_state {
+	struct zcbor_string envelope_str;
+	struct zcbor_string manifest_component_id;
+	uint32_t sequence_number;
+
+	struct suit_integrated_payload integrated_payloads[SUIT_MAX_NUM_INTEGRATED_PAYLOADS];
+	uint_fast32_t integrated_payloads_count;
```
[the remark this answers is not in the document]
Comment (tomchy) [reply include/suit_processor.h:31]: I do not remember one. Probably it was just moved from the original code.
I'll change it to `size_t` to match the remaining definitions.

File: src/suit_manifest.c [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+	}
+
+	if (manifest->components_count >= ZCBOR_ARRAY_SIZE(manifest->component_map)) {
+		SUIT_ERR("Manifest reached the maximum number of components.\r\n");
+		return SUIT_ERR_MANIFEST_VALIDATION;
+	}
+
+	int ret = assign_component_index(component_id, &index);
+	if (ret == SUIT_SUCCESS) {
+		if (components[index].is_dependency == suit_bool_false) {
```
[the remark this answers is not in the document]
Comment (tomchy) [reply src/suit_manifest.c:126]: huh, the trick is that bool is not a boolean, but a special 32 value :slightly_smiling_face: 
In that case we can have:
 - `0` once uninitialized
 - `suit_bool_false` set by `suit_manifest_append_component`
 - `suit_bool_true` set by `suit_manifest_append_dependency`

File: include/suit_types.h (hunk 1 above)
Comment (tomchy) [reply include/suit_types.h:25]: Fixed!

**Review by adsz-nordic — APPROVED** [review]

File: include/suit_processor.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	struct zcbor_string validate_seq;
+	enum suit_seq_status validate_seq_status;
+
+	struct zcbor_string load_seq;
+	enum suit_seq_status load_seq_status;
+
+	struct zcbor_string invoke_seq;
+	enum suit_seq_status invoke_seq_status;
+
+	struct zcbor_string text_seq;
```
Comment (SylwesterKonczyk) [inline include/suit_processor.h:57]: suit-text, being severable, is NOT a sequence (so naming text_seq seems to be appropriate)

File: include/suit_types.h [hunk 5]
```diff
@@ -15,11 +15,14 @@
 
 #define SUIT_MAX_NUM_SIGNERS 2  ///! The maximum number of signers.
 #define SUIT_MAX_NUM_COMPONENT_ID_PARTS 5  ///! The maximum number of bytestrings in a component ID.
-#define SUIT_MAX_NUM_COMPONENTS 4  ///! The maximum number of components referenced in the manifest.
+#define SUIT_MAX_NUM_COMPONENTS 6  ///! The maximum number of components referenced in the manifest.
```
Comment (SylwesterKonczyk) [inline include/suit_types.h:18]: This definition is unclear in a sense that it describes Max number of components per manifest, but naming may suggest it refers to max number of components in entire SUIT tree

**Review by SylwesterKonczyk — APPROVED** [review]

File: include/suit_processor.h (hunk 4 above)
Comment (tomchy) [reply include/suit_processor.h:57]: `_seq` part removed

File: include/suit_types.h (hunk 5 above)
Comment (tomchy) [reply include/suit_types.h:18]: Postponed due to code freeze (requires changes in sdk-ic-next)

**Pull request merged**